api/core/v1alpha1/clustercatalog_types.go

```diff
@@ -98,46 +100,47 @@ type ClusterCatalogStatus struct {
+	// image is the catalog image that backs the content of this catalog.
+	// +optional
 	Image *ImageSource `json:"image,omitempty"`
 }
 
 // ResolvedCatalogSource contains the information about a sourced Catalog
 type ResolvedCatalogSource struct {
-	// Type defines the kind of Catalog content that was sourced.
+	// type defines the kind of Catalog content that was sourced.
 	Type SourceType `json:"type"`
-	// Image is the catalog image that backs the content of this catalog.
-	Image *ResolvedImageSource `json:"image,omitempty"`
+	// image is the catalog image that backs the content of this catalog.
+	Image *ResolvedImageSource `json:"image"`
 }
 
 // ResolvedImageSource contains information about the sourced Catalog
 type ResolvedImageSource struct {
-	// Ref contains the reference to a container image containing Catalog contents.
+	// ref contains the reference to a container image containing Catalog contents.
 	Ref string `json:"ref"`
-	// ResolvedRef contains the resolved sha256 image ref containing Catalog contents.
+	// resolvedRef contains the resolved sha256 image ref containing Catalog contents.
 	ResolvedRef string `json:"resolvedRef"`
-	// LastPollAtempt is the time when the source resolved was last polled for new content.
+	// lastPollAtempt is the time when the source resolved was last polled for new content.
 	LastPollAttempt metav1.Time `json:"lastPollAttempt"`
-	// pullSecret exists to retain compatibility with the existing v1alpha1 APIs. It will be removed in v1alpha2.
-	PullSecret string `json:"pullSecret,omitempty"`
 }
 
 // ImageSource contains information required for sourcing a Catalog from an OCI image
 type ImageSource struct {
-	// Ref contains the reference to a container image containing Catalog contents.
+	// ref contains the reference to a container image containing Catalog contents.
 	Ref string `json:"ref"`
-	// PullSecret contains the name of the image pull secret in the namespace that catalogd is deployed.
+	// pullSecret contains the name of the image pull secret in the namespace that catalogd is deployed.
+	// +optional
 	PullSecret string `json:"pullSecret,omitempty"`
-	// PollInterval indicates the interval at which the image source should be polled for new content,
+	// pollInterval indicates the interval at which the image source should be polled for new content,
 	// specified as a duration (e.g., "5m", "1h", "24h", "etc".). Note that PollInterval may not be
 	// specified for a catalog image referenced by a sha256 digest.
 	// +kubebuilder:validation:Format:=duration
+	// +optional
 	PollInterval *metav1.Duration `json:"pollInterval,omitempty"`
-	// InsecureSkipTLSVerify indicates that TLS certificate validation should be skipped.
-	// If this option is specified, the HTTPS protocol will still be used to
+	// insecureSkipTLSVerify indicates that TLS certificate validation should be skipped.
+	// If this option is specified, the HTTP protocol will still be used to
```

Comment:
This should still be HTTPS, right? Skipping TLS verification means "establish a secure connection with the server, but don't verify its certificate"

```suggestion
	// If this option is specified, the HTTPS protocol will still be used to
```